src/barril/units/_tests/test_array.py

```diff
@@ -328,3 +329,33 @@ def testDefaultValues(unit_database_len):
 
     with pytest.raises(AssertionError):
         Array(ObtainQuantity("m"), unit="m")
+
+
+class TestFromScalar:
+    def test_create_array_informing_category(self):
+        array_molar_fraction = Array.FromScalars(
+            scalars=[Scalar(1, "-"), Scalar(2, "-")], category="percentage"
+        )
+        assert array_molar_fraction.values == [1, 2]
+        assert array_molar_fraction.unit == "-"
+        assert array_molar_fraction.category == "percentage"
+
+    def test_create_array_informing_unit(self):
+        array_in_cm = Array.FromScalars(scalars=[Scalar(1, "m"), Scalar(2, "m")], unit="cm")
```

Comment:
Do you check if all Scalars in the input list have compatible units? (I am asking because I did not find it)

In this line, for example, one scalar has unit `m` and the second one has unit `kg`.

Reply:
💥

Reply:
The `scalar.GetValue(unit)` call would catch this, because it would use the wrong unit:

```python
values = [first_scalar.GetValue(unit)] + [scalar.GetValue(unit) for scalar in scalars]
```

Reply:
Let's add a test for this BTW.